File: validation/duplication_d4_20260819/corpus/features/features.js
import value from "module";

const FIELD = 1;
const concise = item => item + FIELD;
const blockArrow = (item, ...rest) => {
    const first = item + 1;
    const second = first * 2;
    const third = second / 3;
    return third + rest.length;
};

const objectValue = {
    member: 1,
    method(parameter) {
        let first = parameter + 1;
        let second = first * 2;
        let third = second / 3;
        return third;
    },
};

class RoleName {
    static field = 1;
    #privateField = 2;
    static {
        this.field += 1;
        this.field *= 2;
        this.field -= 1;
        this.field ||= 1;
    }
    constructor(value) {
        this.value = value;
        this.value += 1;
        this.value *= 2;
        this.value ??= 0;
    }
    static method(parameter, ...rest) {
        let nilValue = null;
        let truth = true;
        let number = 3.5e2;
        let bigint = 42n;
        let string = "value\n";
        let regex = /a+b?/giu;
        let template = `value ${parameter} ${rest.length}`;
        let RoleName = parameter;

        function nested(shadow) {
            let RoleName = shadow;
            let first = RoleName + 1;
            let second = first * 2;
            return second / 3;
        }

        const anonymous = function (item) {
            const first = item + 1;
            const second = first * 2;
            const third = second / 3;
            return third;
        };

        if (truth && number > 0) {
            number += nested(parameter);
            string += template;
            truth = !truth;
            bigint <<= 1n;
        } else {
            number -= 1;
            string = "fallback";
            truth = false;
            bigint >>= 1n;
        }

        outer: for (let index = 0; index < number; index++) {
            number += index;
            string += index;
            truth ||= index > 1;
            if (index === 2) continue outer;
        }

        for (const key in objectValue) {
            number += key.length;
            string += key;
            truth = truth || key === "member";
            bigint += 1n;
        }

        for (const item of rest) {
            number += item.length;
            string += item;
            truth = truth && item.length > 0;
            bigint += 1n;
        }

        do {
            number += 1;
            bigint += 1n;
            truth = truth && number < 1000;
            string += "d";
        } while (number < 10);

        while (number < 100) {
            number++;
            bigint--;
            truth &&= number !== 0;
            string += ".";
        }

        try {
            number = Number(string);
            bigint += 1n;
            truth = regex.test(string);
            string = string.trim();
        } catch (error) {
            number = -1;
            bigint = 0n;
            truth = false;
            string = error?.message ?? "error";
        } finally {
            number += 1;
            bigint += 1n;
            truth = !truth;
            string = String(number);
        }

        switch (number) {
            case 1:
                number += 1;
                string = "one";
                truth = true;
                break;
            default:
                number = 0;
                string = "default";
                truth = false;
                break;
        }

        {
            number ^= parameter;
            bigint **= 2n;
            truth = truth || false;
            string = objectValue?.method?.(number);
        }
        return { ...objectValue, number, bigint, string, truth, nilValue, regex, anonymous, concise, blockArrow, value };
    }
}

async function* asyncGenerator(stream) {
    let total = 0;
    for await (const item of stream) {
        total += item;
        await stream.checkpoint();
        yield total;
        total = +total;
    }
}

export { RoleName };

File: validation/duplication_d3a_20260818/corpus/features/Features.java
package d0;

import java.util.List;

final class TypeName {}

record CompactRecord(int value) {
    CompactRecord {
        int first = value + 1;
        int second = first + 1;
        int third = second + 1;
        value = third + 1;
    }
}

public class GrammarProbe<T extends TypeName> {
    static final long FIELD = 1L;
    private String member = "member";

    static {
        int first = 1;
        int second = first + 1;
        int third = second + 1;
        int fourth = third + 1;
    }

    {
        member = "one";
        member += "two";
        member = member.trim();
        member = member.toUpperCase();
    }

    public GrammarProbe() {
        this.member = "constructed";
        this.member += "!";
        this.member = this.member.trim();
        this.member = String.valueOf(this.member);
    }

    public synchronized int method(TypeName RoleName, int limit, String... rest) {
        Object nil = null;
        boolean truth = true;
        char character = 'λ';
        String string = "value\n";
        String text = """
                line one
                line two
                """;
        int integer = 0x2A;
        int decimal = 42;
        int octal = 052;
        int binary = 0b101010;
        double floating = 3.5e2;
        float single = 1.25f;
        double hexadecimalFloating = 0x1.0p2;
        long widened = 9L;

        class LocalType {
            int nestedMethod(int value) {
                int first = value + 1;
                int second = first * 2;
                int third = second / 3;
                return third;
            }
        }

        Runnable blockLambda = () -> {
            member = "lambda";
            integer++;
            truth = !truth;
            string += member;
        };
        Runnable expressionLambda = () -> member.trim();
        Runnable anonymous = new Runnable() {
            @Override public void run() {
                member = "anonymous";
                member += "!";
                member = member.trim();
                member = member.toUpperCase();
            }
        };

        if (truth && integer > 0) {
            integer += limit;
            member = string;
            truth = false;
            widened <<= 1;
        } else {
            integer -= limit;
            member = text;
            truth = true;
            widened >>= 1;
        }

        outer: for (int index = 0; index < limit; index++) {
            integer += index;
            member += index;
            truth |= index > 1;
            if (index == 2) continue outer;
        }

        for (String item : rest) {
            integer += item.length();
            member += item;
            truth = truth || item.isEmpty();
            widened += integer;
        }

        do {
            integer += 1;
            widened += 1L;
            truth = truth && integer < 1000;
            member += "d";
        } while (integer < 10);

        while (integer < 100) {
            integer++;
            widened--;
            truth = truth || integer == limit;
            member += ".";
        }

        try {
            integer = Integer.parseInt(member);
            widened += integer;
            truth = integer instanceof Integer;
            member = member.strip();
        } catch (RuntimeException error) {
            integer = -1;
            widened = 0L;
            truth = false;
            member = error.getMessage();
        } finally {
            integer += 1;
            widened += 1L;
            truth = !truth;
            member = String.valueOf(integer);
        }

        try (var resource = new java.io.ByteArrayInputStream(new byte[0])) {
            integer += resource.available();
            widened += integer;
            truth = resource.markSupported();
            member += "resource";
        } catch (java.io.IOException error) {
            integer = -2;
            widened = -2L;
            truth = false;
            member = error.getMessage();
        }

        synchronized (this) {
            integer ^= limit;
            widened >>>= 1;
            truth &= integer != 0;
            member += string;
        }

        {
            integer += 1;
            widened += 1L;
            truth = truth || integer > 0;
            member += "block";
        }

        switch (integer) {
            case 1:
                integer += 1;
                member = "one";
                truth = true;
                break;
            default:
                integer = 0;
                member = "default";
                truth = false;
                break;
        }

        int selected = switch (integer) {
            case 2 -> {
                int first = integer + 1;
                int second = first + 1;
                int third = second + 1;
                yield third + 1;
            }
            default -> 0;
        };
        return integer + selected + character + rest.length + List.of(rest).size();
    }
}
